Tests/Common/Securities/Futures/FuturesExpiryFunctionsTests.cs

```diff
@@ -16,12 +16,114 @@
+using System.Xml;
+using System.Xml.Serialization;
+using System.IO;
 
 namespace QuantConnect.Tests.Common.Securities.Futures
 {
     [TestFixture]
     public class FuturesExpiryFunctionsTests
     {
+        /// <summary>
+        /// Dates for Termination Conditions of futures
+        /// </summary>
+        public class Dates
+        {
+            public DateTime contractMonth;
+            public DateTime lastTrade;
+            public Dates() { }
+            public Dates(DateTime c, DateTime l)
+            {
+                contractMonth = c;
+                lastTrade = l;
+            }
+        }
+
+        /// <summary>
+        /// Symbol and list of dates for testing Futures
+        /// </summary>
+        public class SymbolData
+        {
+            public String symbol;
+            public List<Dates> dateList;
+            public SymbolData() { }
+            public SymbolData(String symbol, List<Dates> list)
+            {
+                this.symbol = symbol;
+                this.dateList = list;
+            }
+        }
+
+        [Test]
+        public void TestAllExpiryDateFunctions()
+        {
+            var _path = Directory.GetCurrentDirectory();
+            _path = _path.Substring(0,_path.Length-10) + "\\Common\\Securities\\Futures\\FuturesExpiryFunctionsTestData.xml";
+            IList<String> symbolsForNineThirtyEasternTime = new List<String>
+            {
+                QuantConnect.Securities.Futures.Indices.NASDAQ100EMini,
+                QuantConnect.Securities.Futures.Indices.SP500EMini,
+                QuantConnect.Securities.Futures.Indices.Dow30EMini
+            };
+            IList<String> symbolsForNineSixteenCentralTime = new List<String>
+            {
+                QuantConnect.Securities.Futures.Currencies.GBP,
+                QuantConnect.Securities.Futures.Currencies.CAD,
+                QuantConnect.Securities.Futures.Currencies.JPY,
+                QuantConnect.Securities.Futures.Currencies.CHF,
+                QuantConnect.Securities.Futures.Currencies.EUR,
+                QuantConnect.Securities.Futures.Currencies.AUD,
+                QuantConnect.Securities.Futures.Currencies.NZD
+            };
+            IList<String> symbolsForTwelveOne = new List<String>
+            {
+                QuantConnect.Securities.Futures.Financials.Y30TreasuryBond,
+                QuantConnect.Securities.Futures.Financials.Y10TreasuryNote,
+                QuantConnect.Securities.Futures.Financials.Y5TreasuryNote,
+                QuantConnect.Securities.Futures.Financials.Y2TreasuryNote
+            };
+            IList<String> symbolsForTwelveOclock = new List<String>
+            {
+                QuantConnect.Securities.Futures.Meats.LiveCattle,
+                QuantConnect.Securities.Futures.Meats.LeanHogs
+            };
+            using (XmlReader reader = XmlReader.Create(_path))
+            {
+                List<SymbolData> data = new List<SymbolData>();
```

Comment:
I have updated it everywhere. And I will keep it in mind from next time onwards.